classes/local/util/manager.php

```diff
@@ -67,9 +67,64 @@ public static function instance($logger) {
     }
 
     /**
+     * Returns a list of (component => name) values.
+     *
+     * @return string[].
+     */
+    public static function get_plugintype_names() {
+
+        $pluginman = \core_plugin_manager::instance();
+        $plugintypes = $pluginman->get_plugin_types();
+
+        // Replacing the directory with the plugin name.
+        foreach ($plugintypes as $type => $dir) {
+            $plugintypes[$type] = $pluginman->plugintype_name($type);
+        }
+
+        return $plugintypes;
+    }
+
+    /**
+     * Returns a list of variables needed by the plugin templates.
+     *
+     * @param string $component.
+     * @return string[].
+     */
+    public static function get_component_variables($component) {
+        $versionvars = \tool_pluginskel\local\skel\version_php_file::get_template_variables();
+        $langvars = \tool_pluginskel\local\skel\lang_file::get_template_variables();
+
+        $templatevars = array_merge($versionvars, $langvars);
+
+        return $templatevars;
+    }
+
+    /**
+     * Returns a list of variables needed by the features' templates.
+     *
+     * @param string $component.
+     * @return string[].
+     */
+    public static function get_features_variables() {
+
+        $featurevars = array();
+
+        // Adding strings.
+        $featurevars[] = array('name' => 'strings', 'hint' => 'multiple', 'values' => array('id', 'text'));
```

Comment:
I don't like this at all. I will change it.

I think the best way to do it would be like this:

```php
array('name'=>'string', 'hint' => 'array', values => array(
    array('name'=>'id', 'hint' => 'boolean', 'required'=>false), // This is just an example.
    array('name'=>'text')
);
```
Each value from the 'values' array would be a full template variable unto itself.